scala/scala_toolchain.bzl

```diff
@@ -0,0 +1,12 @@
+def _scala_toolchain_impl(ctx):
+  toolchain = platform_common.ToolchainInfo(
+    scalacopts = ctx.attr.scalacopts,
+  )
+  return [toolchain]
+
+scala_toolchain = rule(
+  _scala_toolchain_impl,
+  attrs = {
+    'scalacopts': attr.string_list(),
```

Comment:
could we move the scala_library and scala_compiler jars here as well? Since, ultimately, that is how we will be able to switch from 2.11 to 2.12, etc...

Also, we might have default plugins here (since scala native and scala_js use plugins).